src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSMachineWideSettings.cs

```diff
@@ -20,7 +21,8 @@ public VsMachineWideSettings()
         {
             if (_asyncServiceProvider == null)
             {
-                throw new ArgumentNullException(nameof(_asyncServiceProvider));
+                throw new NullReferenceException(
```

Comment:
This one makes me so uneasy. 

NullReferenceException seems like an uncontrolled exception. 

Maybe we can refactor VsMachineWideSettings to have 2 constructors, and the internal constructor can have the argument check.

See https://github.com/NuGet/NuGet.Client/blob/e9722d69400a57d2c0508bbce9e9591fcad550f1/src/NuGet.Clients/NuGet.Console/OutputConsole/OutputConsoleProvider.cs#L29-L40 for example

Reply:
Yeah, I called out the NRE in the PR description as being weird. Though I don't know an alternative exception type to use. I do like this pattern, see if you like the latest commit.

Reply:
yep, looks good to me.